packaging/debian/postinst

```diff
@@ -25,31 +25,82 @@ DAEMON_USER=storj-storagenode
 
 . /usr/share/debconf/confmodule
 
+arch=$(dpkg --print-architecture)
+
+get_binary() {
+    binary=$1
+    tmpfile=$(mktemp /tmp/${binary}.zip.XXXXXX)
+
+    wget -O $tmpfile ${BINARIES_SERVER}/${binary}_linux_${arch}.zip
+    unzip $tmpfile -d /var/lib/storagenode
+    mv /var/lib/storagenode/${binary}_linux_amd64 /var/lib/storagenode/${binary}
+    rm $tmpfile
+}
 
 case "$1" in
     configure)
 	installed_version="$2"
-	#if [ $installed_version != "" ]; then
-	#    echo "This is a reconfiguration or a reinstallation"
-	#fi
+	if [ -n "$installed_version" ]; then
+	    echo "This is a reconfiguration or a reinstallation"
+	fi
+
+	db_get storagenode/identitydir
+	identitydir="$RET"
+	db_get storagenode/wallet
+	wallet="$RET"
+	db_get storagenode/email
+	email="$RET"
+	db_get storagenode/external_addr
+	external_addr="$RET"
+	db_get storagenode/storagedir
+	storagedir="$RET"
+	db_get storagenode/diskspace
+	diskspace="$RET"
 
-	# TODO: should not download binaries if it's only a reconfigure
-	#echo "Downloading binaries..."
 	for binary in "storagenode" "storagenode-updater"; do
-	    wget -O /tmp/${binary}.zip ${BINARIES_SERVER}/${binary}_linux_amd64.zip
-	    unzip /tmp/${binary}.zip -d /var/storagenode/
-	    mv /var/storagenode/${binary}_linux_amd64 /var/storagenode/${binary}
-	    # so that we're in PATH easily
-	    ln -s /var/storagenode/${binary} /usr/bin/${binary}
+	    if [ ! -f "/var/lib/storagenode/${binary}" ]; then
+	        echo "Downloading ${binary} binary..."
+	        get_binary $binary
+	    fi
+
+	    if [ ! -f "/usr/bin/${binary}" ]; then
+			ln -s /var/lib/storagenode/${binary} /usr/bin/${binary}
+	    fi
 	done
-	adduser --system --group --no-create-home --quiet $DAEMON_USER
 
-#	if [ "$old_storagedir" != "" ] && [ "$storagedir" != "$old_storagedir" ]; then
-#	    echo "New storage directory: ${storagedir} (was ${old_storagedir})"
-#	else
-#	    echo "Storage directory: ${storagedir}"
-#	fi
+	# TODO: handle storage dir change during reconfigure
+	if [ -n "$old_storagedir" ] && [ "$storagedir" != "$old_storagedir" ]; then
+	    echo "New storage directory: ${storagedir} (was ${old_storagedir})"
+	else
+	    echo "Storage directory: ${storagedir}"
+	fi
+
+	# copy identity
+	cp ${identitydir}/identity.cert /etc/storagenode/identity
```

Comment:
Can we make these two lines conditional- if the admin has already put a file at `/etc/storagenode/identity/identity.cert`, for example, the package must not overwrite it without asking.